app/models/repository.rb

```diff
@@ -25,6 +28,12 @@ def self.license_options
     presence:   { message: "Project title is required."},
     uniqueness: { message: "Project title is already in use.", scope: :user_username}
 
+  validates :share_type,
```

Comment:
You should also validate that the share_type is included in a list of allowed types (eg: public or private)